src/Components/test/E2ETest/ServerExecutionTests/GlobalizationTest.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            var cultureInfo = CultureInfo.GetCultureInfo(culture);
+
+            var selector = new SelectElement(Browser.FindElement(By.Id("culture-selector")));
+            selector.SelectByValue(culture);
+
+            // That should have triggered a redirect, wait for the main test selector to come up.
+            MountTestComponent<GlobalizationBindCases>();
+            WaitUntilExists(By.Id("globalization-cases"));
+
+            var cultureDisplay = WaitUntilExists(By.Id("culture-name-display"));
+            Assert.Equal($"Culture is: {culture}", cultureDisplay.Text);
+
+            // int
+            var input = Browser.FindElement(By.Id("input_text_int"));
+            var display = Browser.FindElement(By.Id("input_text_int_value"));
+            Browser.Equal(42.ToString(cultureInfo), () => display.Text);
+
+            input.Clear();
+            input.SendKeys(9000.ToString("0,000", cultureInfo));
+            input.SendKeys("\t");
+            Browser.Equal(9000.ToString(cultureInfo), () => display.Text);
+
+            // decimal
+            input = Browser.FindElement(By.Id("input_text_decimal"));
+            display = Browser.FindElement(By.Id("input_text_decimal_value"));
+            Browser.Equal(4.2m.ToString(cultureInfo), () => display.Text);
+
+            input.Clear();
+            input.SendKeys(9000.42m.ToString("0,000.00", cultureInfo));
+            input.SendKeys("\t");
+            Browser.Equal(9000.42m.ToString(cultureInfo), () => display.Text);
+
+            // datetime
+            input = Browser.FindElement(By.Id("input_text_datetime"));
+            display = Browser.FindElement(By.Id("input_text_datetime_value"));
+            Browser.Equal(new DateTime(1985, 3, 4).ToString(cultureInfo), () => display.Text);
+
+            ReplaceText(input, new DateTime(2000, 1, 2).ToString(cultureInfo));
+            input.SendKeys("\t");
+            Browser.Equal(new DateTime(2000, 1, 2).ToString(cultureInfo), () => display.Text);
+
+            // datetimeoffset
+            input = Browser.FindElement(By.Id("input_text_datetimeoffset"));
+            display = Browser.FindElement(By.Id("input_text_datetimeoffset_value"));
+            Browser.Equal(new DateTimeOffset(new DateTime(1985, 3, 4)).ToString(cultureInfo), () => display.Text);
+
+            ReplaceText(input, new DateTimeOffset(new DateTime(2000, 1, 2)).ToString(cultureInfo));
+            input.SendKeys("\t");
+            Browser.Equal(new DateTimeOffset(new DateTime(2000, 1, 2)).ToString(cultureInfo), () => display.Text);
+        }
+
+        // see: https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/214
+        //
+        // Calling Clear() can trigger onchange, which will revert the value to its default.
+        private static void ReplaceText(IWebElement element, string text)
+        {
+            element.SendKeys(Keys.Control + "a");
+            element.SendKeys(text);
+        }
```

Comment:
I'm surprised this problem isn't plaguing us all over with the recent change to how DOM updates works. What's happening here is that when you use `Clear` it will fire `onchange` and set the field to `1/1/0001 ...` then you send the new date value and end up with `1/1/0001 .... 1/2/2000 ...` which isn't valid and gets reset again to `1/1/0001...`.

It makes sense that it behaves this way when you put together all of the pieces, but I've written lots of these tests and its never come up before. Maybe because we're usually working with numbers, and an extra leading `0` is ignored?

Reply:
And yes, I realize I could have made this nullable, but where's the fun in that?

Reply:
I did have to change the E2E bind tests to account for this, because as you point out, `element.Clear()` can no longer truly clear the box if it's bound to a non-nullable number.

I think this is OK because it only affects test cases, not real-world use. In real use, people don't clear a textbox *then* unfocus it *then* refocus it before they start typing. And if they do that, the behavior will still make sense to a human.

Reply:
> Maybe because we're usually working with numbers, and an extra leading 0 is ignored?

I think the reason is that almost all of our tests only bind things to `string` values. It's only the bind and form-specific tests that go through all the different cases with different types of numbers.

Reply:
As a general note, I don't think we had any tests for binding dates before, and I can see why. There's a number of things in this area that are the painful for testing.